src/scripting/game_lua_kernel.cpp

```diff
@@ -3724,6 +3724,19 @@ int game_lua_kernel::intf_get_color_adjust(lua_State *L)
 	return 0;
 }
 
+int game_lua_kernel::intf_screen_fade(lua_State *L)
+{
+	if(game_display_) {
+		color_t fade;
+		fade.r = luaL_checkinteger(L, 1);
+		fade.g = luaL_checkinteger(L, 2);
+		fade.b = luaL_checkinteger(L, 3);
+		fade.a = luaL_checkinteger(L, 4);
+		game_display_->fade_to(fade, luaL_checkinteger(L, 5));
```

Comment:
This API feels quite awkward… 5 integer arguments do not exactly make for an easy-to-understand API.

I think perhaps taking the colour as a table would be easier, so the function only has two arguments. The only thing I'm not sure of is whether the table should be an array-like table (`{r, g, b, a}`) or a key-value table (`{red=r, green=g, blue=b}`).

Also, you need to add a stub with documentation in utils/emmylua/interface.lua.

Reply:
Generally I prefer the latter, since then it's clear what the values in the table represent.

Reply:
I actually kinda like accepting both in this kind of scenario; if I were making an API that _returned_ a colour I'd probably use the `named_tuple` constructor for the result.

For an array-like table, I'd expect it to handle a missing alpha value but not the other components. For a key-value table… I think I'd expect it to handle missing any component.

Reply:
That would be better. I'm not sure how to do it as i haven't used lua or touched the WML interface before, so i just hacked in something that worked.

Reply:
If you're going for a key-value table, I think we have something like `luaW_table_get_default` which should help. For an array table there's `lua_check<std::vector<int>>`.

Reply:
Sorry it commented without showing me the other comments. My only opinion is that using some sort of colour argument would be better than separating the integers, but i'm not confident doing the implementation of any of it. Help is needed here.